xblock/core.py

```diff
@@ -51,15 +52,27 @@ class SharedBlockBase(Plugin):
     Behaviors and attrs which all XBlock like things should share
     """
 
-    resources_dir = 'public'
+    resources_dir = ''
+    public_dir = 'public'
 
     @classmethod
     def get_resources_dir(cls):
+        """
+        Gets the resource directory for this XBlock.
+        """
         return cls.resources_dir
```

Comment:
Sigh. I feel like we should have just had this as a class attr that could have been made a property later... oh well.